File: 12-Classes_and_objects/python_example.cpp
class Student:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    #Method
    def introduce(self):
        print(f"Name: {self.name}, Age: {self.age}")


#Object creation
s1 = Student("Ayesha", 21)

#Call method
s1.introduce()

File: 13-Access_Specifiers/python_example.py
class Student:
    def __init__(self, name, roll, marks):
        self.name = name          #Public
        self._marks = marks       #Protected (convention)
        self.__roll = roll        #Private (name mangling)

    def show_details(self):
        print("Name:", self.name)
        print("Roll Number:", self.__roll)
        print("Marks:", self._marks)


#Object creation
s1 = Student("Firdous", 101, 88)

#Access
print(s1.name)      #Public → Allowed
#print(s1.__roll)  #Private → Error if uncommented

s1.show_details()
